components/server/src/workspace/workspace-starter.ts

```diff
@@ -332,46 +338,123 @@ export class WorkspaceStarter {
         }
     }
 
-    private async buildImageAndStartWorkspace(
-        ctx: TraceContextWithSpan,
-        user: User,
-        workspace: Workspace,
-        instance: WorkspaceInstance,
-        envVars: ResolvedEnvVars,
-    ): Promise<StartWorkspaceResult> {
-        const { span } = ctx;
+    readonly MANAGED_PHASES: WorkspaceInstancePhase[] = ["preparing", "building", "pending"];
 
-        const forceRebuild = !!workspace.context.forceImageBuild;
-        let needsImageBuild: boolean;
-        try {
-            // if we need to build the workspace image we must not wait for actuallyStartWorkspace to return as that would block the
-            // frontend until the image is built.
-            const additionalAuth = await this.getAdditionalImageAuth(envVars);
-            needsImageBuild =
-                forceRebuild || (await this.needsImageBuild(ctx, user, workspace, instance, additionalAuth));
-            if (needsImageBuild) {
-                instance.status.conditions = {
-                    neededImageBuild: true,
-                };
+    public async start(): Promise<Disposable> {
```

Comment:
can we move the reconcilation control loop into its own file/class to reduce the amount of things the workspace-starter takes care of?

Reply:
> own file/class

Own class: that's what I had before: definitely make sense and feels "better", but is way more code changes. If we decided to do that I'd love to discuss the conceptuals first on this PR, and split it up afterwards. :+1: 

Own file: Then we'd need to move 2/3 of the code, which blows of the change even more. + we have not "one thing" (as Anton mentioned yesterday), but two things again....? :thinking:

Reply:
From what I understood, we would move the `start` (becomes job.run) and `controlAllStartWorkspaces` to `workspace-starter-reconiler.ts` (or so). And within that we call `workspaceStarter.buildImageAndStartWorkspace`. Am I missing something?

>  we have not "one thing" (as Anton mentioned yesterday)

I don't think Anton meant to put everything into one file?